qiita_pet/handlers/study_handlers/description_handlers.py

```diff
@@ -107,7 +107,10 @@ def _get_study_and_check_access(self, study_id):
         else:
             check_access(user, study, raise_error=True)
 
-        return study, user
+        full_access = (study.id in user.user_studies | user.shared_studies or
+                       user.level == 'admin')
```

Comment:
check for admin first and return, that way you don't need to do the user database calls if not needed.